app/models/rbac.rb

```diff
@@ -571,6 +571,10 @@ def self.get_belongsto_matches(blist, klass)
         end
       elsif klass == Host
         results.concat(get_belongsto_matches_for_host(vcmeta_list.last))
+      elsif vcmeta_list.last.kind_of?(Host) && klass <= VmOrTemplate
+        host = vcmeta_list.last
+        vms_and_templates = host.send(klass.to_s.tableize).to_a
```

Comment:
@jrafanie yes, you are right `klass.to_s.tableize` was not enough here, because it generated methods like  `manageiq/providers/vmware_infra_manager/vm/vms_and_templates` ....
so I am using base_model. Is this what did you mean ?